tests/unit/bases/test_instance_config.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import pathlib
+import subprocess
+from unittest import mock
+
+import pytest
+
+from craft_providers import Executor, errors
+from craft_providers.bases.errors import BaseConfigurationError
+from craft_providers.bases.instance_config import InstanceConfiguration
+
+
+@pytest.fixture
+def mock_executor():
+    executor_mock = mock.Mock(spec=Executor)
+    yield executor_mock
+
+
+def test_save(mock_executor):
+    config = InstanceConfiguration(compatibility_tag="tag-foo-v1")
+    config_path = pathlib.Path("/etc/crafty-crafty.conf")
+
+    config.save(executor=mock_executor, config_path=config_path)
+
+    assert mock_executor.mock_calls == [
+        mock.call.create_file(
+            destination=config_path, content=mock.ANY, file_mode="0644"
+        )
+    ]
+
+    assert (
+        mock_executor.mock_calls[0].kwargs["content"].read()
+        == b"compatibility_tag: tag-foo-v1\n"
+    )
+
+
+def test_load(mock_executor):
```

Comment:
please add a more descriptive name that test_load (what variant) or a docstring.

Reply:
Fixed